intel_extension_for_transformers/llm/runtime/graph/models/gptneox/gptneox.cpp

```diff
@@ -270,13 +294,20 @@ static bool gptneox_model_eval_internal(model_context& lctx, const model_token*
   {
     auto& logits_out = lctx.logits;
 
+    size_t bs_stride = n_vocab * N;
     if (lctx.logits_all) {
-      logits_out.resize(n_vocab * N);
-      memcpy(logits_out.data(), (float*)ne_get_data(inpL), sizeof(float) * n_vocab * N);
+      logits_out.resize(n_vocab * N * batch_size);
+      for (int i = 0; i < batch_size; ++i) {
+        memcpy(logits_out.data() + i * bs_stride, (float*)ne_get_data(inpL) + (i * bs_stride),
+               sizeof(float) * n_vocab * N);
+      }
     } else {
       // return result for just the last token
-      logits_out.resize(n_vocab);
-      memcpy(logits_out.data(), (float*)ne_get_data(inpL) + (n_vocab * (N - 1)), sizeof(float) * n_vocab);
+      logits_out.resize(n_vocab * batch_size);
+      for (int i = 0; i < batch_size; ++i) {
+        memcpy(logits_out.data() + (i * n_vocab), (float*)ne_get_data(inpL) + (i * bs_stride) + (n_vocab * (N - 1)),
+               sizeof(float) * n_vocab);
+      }
```

Comment:
It only happens in the first tokens. maybe we can add a `slice` kernel before or after `LN`. However, it may have not much acceleration in the "small" prompt (`lm_head` `GEMM` only). But we can try it. cc @a32543254 'cause you asked the same question. We can consider it.

Reply:
you are right!!!!. keep it here for waiting for an optimization PR.